x-pack/test/detection_engine_api_integration/security_and_spaces/tests/create_threat_matching.ts

```diff
@@ -251,6 +251,379 @@ export default ({ getService }: FtrProviderContext) => {
+                  first_seen: '2021-01-26T11:06:03.000Z',
+                  ip: '45.115.45.3',
+                  matched: {
+                    atomic: '45.115.45.3',
+                    field: 'source.ip',
+                    type: 'url',
+                  },
+                  port: 57324,
+                  provider: 'geenensp',
+                  type: 'url',
+                },
+                {
+                  description: 'this should match auditbeat/hosts on ip',
+                  first_seen: '2021-01-26T11:06:03.000Z',
+                  ip: '45.115.45.3',
+                  matched: {
+                    atomic: '45.115.45.3',
+                    field: 'source.ip',
+                    type: 'ip',
+                  },
+                  provider: 'other_provider',
+                  type: 'ip',
+                },
+              ],
+            },
+          ]);
+        });
+
+        it('adds a single indicator that matched multiple fields', async () => {
+          const rule: CreateRulesSchema = {
+            description: 'Detecting root and admin users',
+            name: 'Query with a rule id',
+            severity: 'high',
+            index: ['auditbeat-*'],
+            type: 'threat_match',
+            risk_score: 55,
+            language: 'kuery',
+            rule_id: 'rule-1',
+            from: '1900-01-01T00:00:00.000Z',
+            query: 'source.port: 57324', // narrow our query to a single record that matches two indicators
+            threat_query: 'threat.indicator.ip: *',
+            threat_index: ['filebeat-*'], // Mimics indicators from the filebeat MISP module
+            threat_mapping: [
+              {
+                entries: [
+                  {
+                    value: 'threat.indicator.port',
+                    field: 'source.port',
+                    type: 'mapping',
+                  },
+                ],
+              },
+              {
+                entries: [
+                  {
+                    value: 'threat.indicator.ip',
+                    field: 'source.ip',
+                    type: 'mapping',
+                  },
+                ],
+              },
+            ],
+            threat_filters: [],
+          };
+
+          const { id } = await createRule(supertest, rule);
+          await waitForRuleSuccessOrStatus(supertest, id);
+          await waitForSignalsToBePresent(supertest, 1, [id]);
+          const signalsOpen = await getSignalsByIds(supertest, [id]);
+          expect(signalsOpen.hits.hits.length).equal(1);
+
+          const { hits } = signalsOpen.hits;
+          const threats = hits.map((hit) => hit._source.threat);
+          // TODO how should a signal be enriched if a single indicator matches
+          // on multiple fields? As evidenced below, we currently duplicate the
+          // indicator with a different matched object.
+          expect(threats).to.eql([
```

Comment:
I wanted to call this out as another behavioral edge case: We do not merge matched indicators during enrichment, so in certain circumstances a given indicator document could appear multiple times in an enriched alert (albeit with different `threat.indicator.matched` data).